--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -121,6 +_,7 @@
 import net.minecraft.util.debugchart.SampleLogger;
 import net.minecraft.util.debugchart.TpsDebugDimensions;
 import net.minecraft.util.profiling.EmptyProfileResults;
+import net.minecraft.util.profiling.InactiveProfiler;
 import net.minecraft.util.profiling.ProfileResults;
 import net.minecraft.util.profiling.Profiler;
 import net.minecraft.util.profiling.ProfilerFiller;
@@ -1289,9 +_,9 @@
             }
             // Paper end - Add onboarding message for initial server start
             // Paper start - Improve outdated version checking
-            if (System.getProperty("paper.disableStartupVersionCheck") == null && io.papermc.paper.configuration.GlobalConfiguration.get().updateChecker.enabled) {
+            /*if (System.getProperty("paper.disableStartupVersionCheck") == null && io.papermc.paper.configuration.GlobalConfiguration.get().updateChecker.enabled) {
                 CompletableFuture.runAsync(com.destroystokyo.paper.PaperVersionFetcher::getUpdateStatusStartupMessage, io.papermc.paper.util.MCUtil.ASYNC_EXECUTOR);
-            }
+            }*/ // Grinder - no startup update message
             // Paper end - Improve outdated version checking
 
             while (this.running) {
@@ -2710,6 +_,7 @@
     // CraftBukkit end
 
     private ProfilerFiller createProfiler() {
+        if (true) return InactiveProfiler.INSTANCE;
         if (this.willStartRecordingMetrics) {
             this.metricsRecorder = ActiveMetricsRecorder.createStarted(
                 new ServerMetricsSamplersProvider(Util.timeSource, this.isDedicatedServer()),
