--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -331,7 +_,7 @@
         }
         public int maxJoinsPerTick = 5;
         public boolean sendFullPosForItemEntities = false;
-        public boolean loadPermissionsYmlBeforePlugins = true;
+        //public boolean loadPermissionsYmlBeforePlugins = true; // Grinder - remove permisions.yml
         @Constraints.Min(4)
         public int regionFileCacheSize = 256;
         @Comment("See https://luckformula.emc.gs")
