From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Tue, 6 Jan 2026 17:40:55 +0100
Subject: [PATCH] deactivate permissions.yml


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index b2961752b50e98d95adf3af4f658e2fdb2127d43..773db43700927461c7db62b46bd14997e5e8b8c9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1121,7 +1121,7 @@ public final class CraftServer implements Server {
 
     @SuppressWarnings({ "unchecked", "finally" })
     private void loadCustomPermissions() {
-        File file = new File(this.configuration.getString("settings.permissions-file"));
+        /*File file = new File(this.configuration.getString("settings.permissions-file"));
         FileInputStream stream;
 
         try {
@@ -1163,7 +1163,7 @@ public final class CraftServer implements Server {
             } catch (IllegalArgumentException ex) {
                 this.getLogger().log(Level.SEVERE, "Permission in " + file + " was already defined", ex);
             }
-        }
+        }*/ // Grinder
     }
 
     @Override
diff --git a/src/main/resources/configurations/bukkit.yml b/src/main/resources/configurations/bukkit.yml
index 00a5b4dc0d7bc57ae3f9231f84a081617ec4b15b..629a61cb26543cc3d32e875ecac3920b133949f3 100644
--- a/src/main/resources/configurations/bukkit.yml
+++ b/src/main/resources/configurations/bukkit.yml
@@ -12,7 +12,6 @@
 settings:
     allow-end: true
     warn-on-overload: true
-    permissions-file: permissions.yml
     update-folder: update
     plugin-profiling: false
     connection-throttle: 4000
