From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Tue, 6 Jan 2026 20:02:03 +0100
Subject: [PATCH] remove commands.yml


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 773db43700927461c7db62b46bd14997e5e8b8c9..65c060f13811df86c6de3d17bf0c22df9f03294a 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -279,7 +279,7 @@ public final class CraftServer implements Server {
     private final DedicatedPlayerList playerList;
     private final Map<String, World> worlds = new LinkedHashMap<>();
     private YamlConfiguration configuration;
-    private YamlConfiguration commandsConfiguration;
+    //private YamlConfiguration commandsConfiguration; // Grinder
     private final Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
     private final Map<UUID, OfflinePlayer> offlinePlayers = new MapMaker().weakValues().makeMap();
     private final EntityMetadataStore entityMetadata = new EntityMetadataStore();
@@ -435,7 +435,7 @@ public final class CraftServer implements Server {
             this.configuration.set("aliases", "now-in-commands.yml");
         }
         this.saveConfig();
-        if (this.getCommandsConfigFile().isFile()) {
+        /*if (this.getCommandsConfigFile().isFile()) {
             legacyAlias = null;
         }
         this.commandsConfiguration = YamlConfiguration.loadConfiguration(this.getCommandsConfigFile());
@@ -466,9 +466,9 @@ public final class CraftServer implements Server {
             }
         }
 
-        this.saveCommandsConfig();
+        this.saveCommandsConfig();x
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
-        this.ignoreVanillaPermissions = this.commandsConfiguration.getBoolean("ignore-vanilla-permissions");
+        this.ignoreVanillaPermissions = this.commandsConfiguration.getBoolean("ignore-vanilla-permissions");*/ // Grinder
         this.overrideSpawnLimits();
         console.autosavePeriod = this.configuration.getInt("ticks-per.autosave");
         this.warningState = WarningState.value(this.configuration.getString("settings.deprecated-verbose"));
@@ -488,16 +488,16 @@ public final class CraftServer implements Server {
     }
 
     public boolean getCommandBlockOverride(String command) {
-        return this.overrideAllCommandBlockCommands || this.commandsConfiguration.getStringList("command-block-overrides").contains(command);
+        return /*this.overrideAllCommandBlockCommands || this.commandsConfiguration.getStringList("command-block-overrides").contains(command)*/ false; // Grinder
     }
 
     private File getConfigFile() {
         return (File) this.console.options.valueOf("bukkit-settings");
     }
 
-    private File getCommandsConfigFile() {
+    /*private File getCommandsConfigFile() {
         return (File) this.console.options.valueOf("commands-settings");
-    }
+    }*/ // Grinder
 
     private void overrideSpawnLimits() {
         for (SpawnCategory spawnCategory : SpawnCategory.values()) {
@@ -515,13 +515,13 @@ public final class CraftServer implements Server {
         }
     }
 
-    private void saveCommandsConfig() {
+    /*private void saveCommandsConfig() {
         try {
             this.commandsConfiguration.save(this.getCommandsConfigFile());
         } catch (IOException ex) {
             Logger.getLogger(CraftServer.class.getName()).log(Level.SEVERE, "Could not save " + this.getCommandsConfigFile(), ex);
         }
-    }
+    }*/ // Grinder
 
     private void loadCompatibilities() {
         if (true) return; // Paper - Big nope
@@ -955,7 +955,7 @@ public final class CraftServer implements Server {
 
     @Override
     public void reload() {
-        // Paper start - lifecycle events
+        /*// Paper start - lifecycle events
         if (io.papermc.paper.plugin.lifecycle.event.LifecycleEventRunner.INSTANCE.blocksPluginReloading()) {
             throw new IllegalStateException(org.bukkit.command.defaults.ReloadCommand.RELOADING_DISABLED_MESSAGE);
         }
@@ -1061,7 +1061,7 @@ public final class CraftServer implements Server {
         this.syncCommands(); // Refresh commands after event
         // Paper end - brigadier command API
         new ServerLoadEvent(ServerLoadEvent.LoadType.RELOAD).callEvent();
-        org.spigotmc.WatchdogThread.hasStarted = true; // Paper - Disable watchdog early timeout on reload
+        org.spigotmc.WatchdogThread.hasStarted = true; // Paper - Disable watchdog early timeout on reload*/ // Grinder
     }
 
     // Paper start - Wait for Async Tasks during shutdown
@@ -1672,7 +1672,7 @@ public final class CraftServer implements Server {
 
     @Override
     public Map<String, String[]> getCommandAliases() {
-        ConfigurationSection section = this.commandsConfiguration.getConfigurationSection("aliases");
+        /*ConfigurationSection section = this.commandsConfiguration.getConfigurationSection("aliases");
         Map<String, String[]> result = new LinkedHashMap<>();
 
         if (section != null) {
@@ -1689,7 +1689,7 @@ public final class CraftServer implements Server {
             }
         }
 
-        return result;
+        return result;*/ return Map.of(); // Grinder
     }
 
     @Override
@@ -2866,7 +2866,7 @@ public final class CraftServer implements Server {
                 .map(key -> key.toLowerCase(java.util.Locale.ENGLISH))
                 .collect(java.util.stream.Collectors.toSet());
         getCommandMap().getKnownCommands().keySet().removeIf(removals::contains);
-        File file = getCommandsConfigFile();
+        /*File file = getCommandsConfigFile();
         try {
             commandsConfiguration.load(file);
         } catch (FileNotFoundException ex) {
@@ -2874,7 +2874,7 @@ public final class CraftServer implements Server {
         } catch (IOException | org.bukkit.configuration.InvalidConfigurationException ex) {
             Bukkit.getLogger().log(Level.SEVERE, "Cannot load " + file, ex);
             return false;
-        }
+        }*/
         commandMap.registerServerAliases();
         return true;
     }
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 9d4d97fcb0f3cc94e9a263c61cc0e7e58fd322c3..68cb3c153ae28327c5231c7560dad02d32f22bd0 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -117,11 +117,11 @@ public class Main {
                         .defaultsTo(new File("bukkit.yml"))
                         .describedAs("Yml file");
 
-                this.acceptsAll(asList("C", "commands-settings"), "File for command settings")
+                /*this.acceptsAll(asList("C", "commands-settings"), "File for command settings")
                         .withRequiredArg()
                         .ofType(File.class)
                         .defaultsTo(new File("commands.yml"))
-                        .describedAs("Yml file");
+                        .describedAs("Yml file");*/ // Grinder
 
                 this.accepts("forceUpgrade", "Whether to force a world upgrade");
                 this.accepts("eraseCache", "Whether to force cache erase during world upgrade");
diff --git a/src/main/resources/configurations/commands.yml b/src/main/resources/configurations/commands.yml
deleted file mode 100644
index ddf2cd4b802cf7d22342eed7f09c989761aed7a9..0000000000000000000000000000000000000000
--- a/src/main/resources/configurations/commands.yml
+++ /dev/null
@@ -1,15 +0,0 @@
-# This is the Bukkit commands configuration file for Paper.
-#
-# If you need help with the configuration or have any questions related to Paper,
-# join us in our Discord or check the docs page.
-#
-# File Reference: https://docs.papermc.io/paper/reference/bukkit-commands-configuration/
-# Docs: https://docs.papermc.io/
-# Discord: https://discord.gg/papermc
-# Website: https://papermc.io/
-
-command-block-overrides: []
-ignore-vanilla-permissions: false
-aliases:
-    icanhasbukkit:
-      - "version $1-"
