--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -14,6 +_,7 @@
 import java.util.Objects;
 import java.util.Set;
 import java.util.logging.Level;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.resources.Identifier;
 import net.minecraft.server.MinecraftServer;
@@ -151,9 +_,9 @@
         SpigotConfig.sendNamespaced = SpigotConfig.getBoolean("commands.send-namespaced", true);
     }
 
-    public static String whitelistMessage;
-    public static String unknownCommandMessage;
-    public static String serverFullMessage;
+    public static String whitelistMessage = ""; // Grinder
+    public static String unknownCommandMessage = ""; // Grinder
+    /*public static String serverFullMessage;
     public static String outdatedClientMessage = "Outdated client! Please use {0}";
     public static String outdatedServerMessage = "Outdated server! I'm still on {0}";
 
@@ -172,7 +_,7 @@
         SpigotConfig.serverFullMessage = SpigotConfig.transform(SpigotConfig.getString("messages.server-full", "The server is full!"));
         SpigotConfig.outdatedClientMessage = SpigotConfig.transform(SpigotConfig.getString("messages.outdated-client", SpigotConfig.outdatedClientMessage));
         SpigotConfig.outdatedServerMessage = SpigotConfig.transform(SpigotConfig.getString("messages.outdated-server", SpigotConfig.outdatedServerMessage));
-    }
+    }*/ // Grinder - use mojang's messages
 
     public static int timeoutTime = 60;
     public static boolean restartOnCrash = true;
@@ -182,8 +_,8 @@
         SpigotConfig.timeoutTime = SpigotConfig.getInt("settings.timeout-time", SpigotConfig.timeoutTime);
         SpigotConfig.restartOnCrash = SpigotConfig.getBoolean("settings.restart-on-crash", SpigotConfig.restartOnCrash);
         SpigotConfig.restartScript = SpigotConfig.getString("settings.restart-script", SpigotConfig.restartScript);
-        SpigotConfig.restartMessage = SpigotConfig.transform(SpigotConfig.getString("messages.restart", "Server is restarting"));
-        SpigotConfig.commands.put("restart", new RestartCommand("restart"));
+        /*SpigotConfig.restartMessage = SpigotConfig.transform(SpigotConfig.getString("messages.restart", "Server is restarting"));
+        SpigotConfig.commands.put("restart", new RestartCommand("restart"));*/ // Grinder
     }
 
     public static boolean bungee;
