--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -538,7 +_,7 @@
     }
     // Paper end - PlayerLoginEvent
     public LoginResult canPlayerLogin(SocketAddress socketAddress, NameAndId nameAndId) { // Paper - PlayerLoginEvent
-        LoginResult whitelistEventResult; // Paper
+        /*LoginResult whitelistEventResult; // Paper // Grinder
         // Paper start - Fix MC-158900
         UserBanListEntry userBanListEntry;
         if (this.bans.isBanned(nameAndId) && (userBanListEntry = this.bans.get(nameAndId)) != null) {
@@ -565,9 +_,9 @@
             }
 
             return new LoginResult(mutableComponent, org.bukkit.event.player.PlayerLoginEvent.Result.KICK_BANNED); // Paper - PlayerLoginEvent
-        } else {
+        } else {*/ // Grinder
             return this.canBypassFullServerLogin(nameAndId, new LoginResult(Component.translatable("multiplayer.disconnect.server_full"), org.bukkit.event.player.PlayerLoginEvent.Result.KICK_FULL)); // Paper - PlayerServerFullCheckEvent
-        }
+        //} // Grinder
     }
 
     public boolean disconnectAllPlayersWithProfile(GameProfile profile) { // Paper - validate usernames
@@ -733,13 +_,14 @@
         this.sendPlayerPermissionLevel(player, profilePermissions, recalculatePermissions); // Paper - avoid recalculating permissions if possible
     }
 
+    private static final EnumSet<ClientboundPlayerInfoUpdatePacket.Action> UPDATE_LATENCY = EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY); // Grinder
     public void tick() {
         if (++this.sendAllPlayerInfoIn > 600) {
             // CraftBukkit start
             for (int i = 0; i < this.players.size(); ++i) {
                 final ServerPlayer target = this.players.get(i);
 
-                target.connection.send(new ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY), com.google.common.collect.Collections2.filter(this.players, t -> target.getBukkitEntity().canSee(t.getBukkitEntity()))));
+                target.connection.send(new ClientboundPlayerInfoUpdatePacket(UPDATE_LATENCY, com.google.common.collect.Collections2.filter(this.players, t -> target.getBukkitEntity().canSee(t.getBukkitEntity()))));
             }
             // CraftBukkit end
             this.sendAllPlayerInfoIn = 0;
