From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Thu, 30 May 2024 22:41:10 +0200
Subject: [PATCH] deactivate-permissions-yml


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index c7df339aeb62ee627edaf1bb4c8474b61e357ba6..b18bde69ec71626284ce75e05bdcf3af2cc59805 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1201,7 +1201,7 @@ public final class CraftServer implements Server {
 
     @SuppressWarnings({ "unchecked", "finally" })
     private void loadCustomPermissions() {
-        File file = new File(this.configuration.getString("settings.permissions-file"));
+        /*File file = new File(this.configuration.getString("settings.permissions-file"));
         FileInputStream stream;
 
         try {
@@ -1243,7 +1243,7 @@ public final class CraftServer implements Server {
             } catch (IllegalArgumentException ex) {
                 this.getLogger().log(Level.SEVERE, "Permission in " + file + " was already defined", ex);
             }
-        }
+        }*/ // Grinder - deactivate permissions.yml
     }
 
     @Override
diff --git a/src/main/resources/configurations/bukkit.yml b/src/main/resources/configurations/bukkit.yml
index eef7c125b2689f29cae5464659eacdf33f5695b2..ffa67adc26f087cc991b7f59ef1363843849f5f4 100644
--- a/src/main/resources/configurations/bukkit.yml
+++ b/src/main/resources/configurations/bukkit.yml
@@ -14,7 +14,6 @@
 settings:
     allow-end: true
     warn-on-overload: true
-    permissions-file: permissions.yml
     update-folder: update
     plugin-profiling: false
     connection-throttle: 4000
