From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Fri, 31 May 2024 00:32:13 +0200
Subject: [PATCH] only init keypair when online mode is true


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 78c2fcceced6f4f156ec9dc8b8046a9ac0c7ab4a..4ddc97f132245951b8d5f2d6dba98e3dac1baf7a 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -272,7 +272,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         }
         // Paper end - Unix domain socket support
 
-        this.initializeKeyPair();
+        if (this.usesAuthentication()) this.initializeKeyPair(); // Grinder - only init when online mode is true
         DedicatedServer.LOGGER.info("Starting Minecraft server on {}:{}", this.getLocalIp().isEmpty() ? "*" : this.getLocalIp(), this.getPort());
 
         try {
