From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Sat, 1 Jun 2024 14:33:58 +0200
Subject: [PATCH] optimize playerlist ticking


diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
index 9ccca41bf23efadba5329cc584bbcdcacbe09a92..bc67d60e0567e9c5a0a42569692801afbb1a1212 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket.java
@@ -190,7 +190,7 @@ public class ClientboundPlayerInfoUpdatePacket implements Packet<ClientGamePacke
         @Nullable Component displayName,
         @Nullable RemoteChatSession.Data chatSession
     ) {
-        Entry(ServerPlayer player) {
+        public Entry(ServerPlayer player) { // Grinder - public
             // Paper start - Add Listing API for Player
             this(player, true);
         }
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 791dd164ba7f416970b43ad74af4ec86087787dc..289adf5b4d0f75a68d91f5fd16622d240899985b 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -977,18 +977,29 @@ public abstract class PlayerList {
         this.sendPlayerPermissionLevel(player, i, recalculatePermissions); // Paper - avoid recalculating permissions if possible
     }
 
+    private static final EnumSet<ClientboundPlayerInfoUpdatePacket.Action> LATENCY_ACTION = EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY); // Grinder
     public void tick() {
         if (++this.sendAllPlayerInfoIn > 600) {
             // CraftBukkit start
             for (int i = 0; i < this.players.size(); ++i) {
                 final ServerPlayer target = (ServerPlayer) this.players.get(i);
 
-                target.connection.send(new ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY), this.players.stream().filter(new Predicate<ServerPlayer>() {
+                // Grinder Start
+                final List<ClientboundPlayerInfoUpdatePacket.Entry> entries = new java.util.ArrayList<>();
+                for (int y = 0; y < this.players.size(); y++) {
+                    final ServerPlayer player = this.players.get(y);
+                    if (target.getBukkitEntity().canSee(player.getBukkitEntity())) {
+                        entries.add(new ClientboundPlayerInfoUpdatePacket.Entry(player));
+                    }
+                }
+                target.connection.send(new ClientboundPlayerInfoUpdatePacket(LATENCY_ACTION, entries));
+                // Grinder - Stop
+                /*target.connection.send(new ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY), this.players.stream().filter(new Predicate<ServerPlayer>() {
                     @Override
                     public boolean test(ServerPlayer input) {
                         return target.getBukkitEntity().canSee(input.getBukkitEntity());
                     }
-                }).collect(Collectors.toList())));
+                }).collect(Collectors.toList())));*/ // Grinder - optimizing
             }
             // CraftBukkit end
             this.sendAllPlayerInfoIn = 0;
