From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Fri, 31 May 2024 15:02:39 +0200
Subject: [PATCH] remove use of HelpYamlReader


diff --git a/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java b/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
index 97141968f36b3ef88bd6e520c2ccc37c97e4adb1..55ef9d081ff87c6e94f2a3b493fc3261110371c3 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/SimpleHelpMap.java
@@ -36,19 +36,19 @@ public class SimpleHelpMap implements HelpMap {
     private final Map<String, HelpTopic> helpTopics;
     private final Map<Class, HelpTopicFactory<Command>> topicFactoryMap;
     private final CraftServer server;
-    private HelpYamlReader yaml;
+    // private HelpYamlReader yaml; // Grinder
 
     @SuppressWarnings("unchecked")
     public SimpleHelpMap(CraftServer server) {
         this.helpTopics = new TreeMap<String, HelpTopic>(HelpTopicComparator.topicNameComparatorInstance()); // Using a TreeMap for its explicit sorting on key
         this.topicFactoryMap = new HashMap<Class, HelpTopicFactory<Command>>();
         this.server = server;
-        this.yaml = new HelpYamlReader(server);
+        //this.yaml = new HelpYamlReader(server); // Grinder
 
         Predicate indexFilter = Predicates.not(Predicates.instanceOf(CommandAliasHelpTopic.class));
-        if (!this.yaml.commandTopicsInMasterIndex()) {
+        /*if (!this.yaml.commandTopicsInMasterIndex()) {
             indexFilter = Predicates.and(indexFilter, Predicates.not(new IsCommandTopicPredicate()));
-        }
+        }*/ // Grinder
 
         this.defaultTopic = new IndexHelpTopic("Index", null, null, Collections2.filter(this.helpTopics.values(), indexFilter), "Use /help [n] to get page n of help.");
 
@@ -88,14 +88,15 @@ public class SimpleHelpMap implements HelpMap {
 
     @Override
     public List<String> getIgnoredPlugins() {
-        return this.yaml.getIgnoredPlugins();
+        // return this.yaml.getIgnoredPlugins();
+        return List.of(); // Grinder
     }
 
     /**
      * Reads the general topics from help.yml and adds them to the help index.
      */
     public synchronized void initializeGeneralTopics() {
-        this.yaml = new HelpYamlReader(this.server);
+        /*this.yaml = new HelpYamlReader(this.server);
 
         // Initialize general help topics from the help.yml file
         for (HelpTopic topic : this.yaml.getGeneralTopics()) {
@@ -109,7 +110,7 @@ public class SimpleHelpMap implements HelpMap {
             } else {
                 this.addTopic(topic);
             }
-        }
+        }*/ // Grinder
     }
 
     /**
@@ -117,7 +118,8 @@ public class SimpleHelpMap implements HelpMap {
      */
     public synchronized void initializeCommands() {
         // ** Load topics from highest to lowest priority order **
-        Set<String> ignoredPlugins = new HashSet<String>(this.yaml.getIgnoredPlugins());
+        //Set<String> ignoredPlugins = new HashSet<String>(this.yaml.getIgnoredPlugins());
+        final Set<String> ignoredPlugins = Set.of(); // Grinder
 
         // Don't load any automatic help topics if All is ignored
         if (ignoredPlugins.contains("All")) {
@@ -174,14 +176,14 @@ public class SimpleHelpMap implements HelpMap {
         }
 
         // Amend help topics from the help.yml file
-        for (HelpTopicAmendment amendment : this.yaml.getTopicAmendments()) {
+        /*for (HelpTopicAmendment amendment : this.yaml.getTopicAmendments()) {
             if (this.helpTopics.containsKey(amendment.getTopicName())) {
                 this.helpTopics.get(amendment.getTopicName()).amendTopic(amendment.getShortText(), amendment.getFullText());
                 if (amendment.getPermission() != null) {
                     this.helpTopics.get(amendment.getTopicName()).amendCanSee(amendment.getPermission());
                 }
             }
-        }
+        }*/ // Grinder
     }
 
     private void fillPluginIndexes(Map<String, Set<HelpTopic>> pluginIndexes, Collection<? extends Command> commands) {
diff --git a/src/main/resources/configurations/help.yml b/src/main/resources/configurations/help.yml
deleted file mode 100644
index 15c3d07070624ad453032a02d8d4d3f26bd6cb8d..0000000000000000000000000000000000000000
--- a/src/main/resources/configurations/help.yml
+++ /dev/null
@@ -1,55 +0,0 @@
-# This is the help configuration file for Bukkit.
-# 
-# By default you do not need to modify this file. Help topics for all plugin commands are automatically provided by
-# or extracted from your installed plugins. You only need to modify this file if you wish to add new help pages to
-# your server or override the help pages of existing plugin commands.
-# 
-# This file is divided up into the following parts:
-# -- general-topics: lists admin defined help topics
-# -- index-topics:   lists admin defined index topics
-# -- amend-topics:   lists topic amendments to apply to existing help topics
-# -- ignore-plugins: lists any plugins that should be excluded from help
-# 
-# Examples are given below. When amending command topic, the string <text> will be replaced with the existing value
-# in the help topic. Color codes can be used in topic text. The color code character is & followed by 0-F.
-# ================================================================
-# 
-# Set this to true to list the individual command help topics in the master help.
-# command-topics-in-master-index: true
-# 
-# Each general topic will show up as a separate topic in the help index along with all the plugin command topics.
-# general-topics:
-#     Rules:
-#         shortText: Rules of the server
-#         fullText: |
-#             &61. Be kind to your fellow players.
-#             &B2. No griefing.
-#             &D3. No swearing.
-#         permission: topics.rules
-# 
-# Each index topic will show up as a separate sub-index in the help index along with all the plugin command topics.
-# To override the default help index (displayed when the user executes /help), name the index topic "Default".
-# index-topics:
-#     Ban Commands:
-#         shortText: Player banning commands
-#         preamble: Moderator - do not abuse these commands
-#         permission: op
-#         commands:
-#             - /ban
-#             - /ban-ip
-#             - /banlist
-# 
-# Topic amendments are used to change the content of automatically generated plugin command topics.
-# amended-topics:
-#     /stop:
-#         shortText: Stops the server cold....in its tracks!
-#         fullText: <text> - This kills the server.
-#         permission: you.dont.have
-# 
-# Any plugin in the ignored plugins list will be excluded from help. The name must match the name displayed by
-# the /plugins command. Ignore "Bukkit" to remove the standard bukkit commands from the index. Ignore "All"
-# to completely disable automatic help topic generation.
-# ignore-plugins:
-#    - PluginNameOne
-#    - PluginNameTwo
-#    - PluginNameThree
