From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Fri, 31 May 2024 15:11:24 +0200
Subject: [PATCH] tweak default configs


diff --git a/src/main/java/net/minecraft/server/ServerAdvancementManager.java b/src/main/java/net/minecraft/server/ServerAdvancementManager.java
index 2172e23b22b411fe79e930128bd43ac07becb620..5b1b60112f9ae6abf96d5f04d57e096759805dca 100644
--- a/src/main/java/net/minecraft/server/ServerAdvancementManager.java
+++ b/src/main/java/net/minecraft/server/ServerAdvancementManager.java
@@ -48,9 +48,10 @@ public class ServerAdvancementManager extends SimpleJsonResourceReloadListener {
         RegistryOps<JsonElement> registryops = this.registries.createSerializationContext(JsonOps.INSTANCE);
         Builder<ResourceLocation, AdvancementHolder> builder = ImmutableMap.builder();
 
+        if (org.spigotmc.SpigotConfig.disabledAdvancements != null && !org.spigotmc.SpigotConfig.disabledAdvancements.contains("*")) { // Grinder - move on top, to only check once
         prepared.forEach((minecraftkey, jsonelement) -> {
             // Spigot start
-            if (org.spigotmc.SpigotConfig.disabledAdvancements != null && (org.spigotmc.SpigotConfig.disabledAdvancements.contains("*") || org.spigotmc.SpigotConfig.disabledAdvancements.contains(minecraftkey.toString()) || org.spigotmc.SpigotConfig.disabledAdvancements.contains(minecraftkey.getNamespace()))) {
+            if (org.spigotmc.SpigotConfig.disabledAdvancements.contains(minecraftkey.toString()) || org.spigotmc.SpigotConfig.disabledAdvancements.contains(minecraftkey.getNamespace())) {
                 return;
             }
             // Spigot end
@@ -64,7 +65,7 @@ public class ServerAdvancementManager extends SimpleJsonResourceReloadListener {
                 ServerAdvancementManager.LOGGER.error("Parsing error loading custom advancement {}: {}", minecraftkey, exception.getMessage());
             }
 
-        });
+        });}
         this.advancements = new HashMap<>(builder.buildOrThrow()); // CraftBukkit - SPIGOT-7734: mutable
         AdvancementTree advancementtree = new AdvancementTree();
 
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
index d6431376184e5650b370cbab204e28bc31f4dac6..ec1320941e8a4004f33182c0a2af1f13a0abe8f8 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -127,7 +127,7 @@ public class DedicatedServerProperties extends Settings<DedicatedServerPropertie
         this.rconPort = this.get("rcon.port", 25575);
         this.rconPassword = this.get("rcon.password", "");
         this.hardcore = this.get("hardcore", false);
-        this.allowNether = this.get("allow-nether", true);
+        this.allowNether = this.get("allow-nether", false);
         this.spawnMonsters = this.get("spawn-monsters", true);
         this.useNativeTransport = this.get("use-native-transport", true);
         this.enableCommandBlock = this.get("enable-command-block", false);
diff --git a/src/main/java/org/spigotmc/SpigotConfig.java b/src/main/java/org/spigotmc/SpigotConfig.java
index 2bbee0823cef128a5c2cc2dfb67c403a4ba96bfc..a1999eb8e364a75688132cc4086af44f8f1b606b 100644
--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -388,8 +388,8 @@ public class SpigotConfig
     public static boolean disableAdvancementSaving;
     public static List<String> disabledAdvancements;
     private static void disabledAdvancements() {
-        SpigotConfig.disableAdvancementSaving = SpigotConfig.getBoolean("advancements.disable-saving", false);
-        SpigotConfig.disabledAdvancements = SpigotConfig.getList("advancements.disabled", Arrays.asList(new String[]{"minecraft:story/disabled"}));
+        SpigotConfig.disableAdvancementSaving = SpigotConfig.getBoolean("advancements.disable-saving", true); // Grinder - tweak default configs
+        SpigotConfig.disabledAdvancements = SpigotConfig.getList("advancements.disabled", List.of("*")); // Grinder - tweak default configs
     }
 
     public static boolean logVillagerDeaths;
diff --git a/src/main/resources/configurations/bukkit.yml b/src/main/resources/configurations/bukkit.yml
index 3c05f7c3fba89ef0572bbb7d8a0d73365b115849..25aaf35cb1728f58ddb6e79e94b53f3d4923fb9a 100644
--- a/src/main/resources/configurations/bukkit.yml
+++ b/src/main/resources/configurations/bukkit.yml
@@ -12,7 +12,7 @@
 
 
 settings:
-    allow-end: true
+    allow-end: false
     warn-on-overload: true
     update-folder: update
     plugin-profiling: false
