From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Sat, 1 Jun 2024 15:41:48 +0200
Subject: [PATCH] configuration


diff --git a/src/main/java/grinder/GrinderConfiguration.java b/src/main/java/grinder/GrinderConfiguration.java
new file mode 100644
index 0000000000000000000000000000000000000000..e70456fee4f6b4eace453555c6d9b35bc07c4bc4
--- /dev/null
+++ b/src/main/java/grinder/GrinderConfiguration.java
@@ -0,0 +1,26 @@
+package grinder;
+
+import io.papermc.paper.configuration.ConfigurationLoaders;
+import java.nio.file.Path;
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.ConfigurateException;
+import org.spongepowered.configurate.yaml.YamlConfigurationLoader;
+
+public final class GrinderConfiguration {
+    public static final boolean SERVER_SIDE_FROZEN;
+    public static final boolean BLOCK_ENTITY_TICKING;
+
+    static {
+        try {
+            final YamlConfigurationLoader loader = ConfigurationLoaders.naturallySorted()
+                .path(Path.of("config", "grinder.yml"))
+                .build();
+            final CommentedConfigurationNode node = loader.load();
+            SERVER_SIDE_FROZEN = node.node("server-side-frozen").getBoolean(false);
+            BLOCK_ENTITY_TICKING = node.node("block-entity-ticking").getBoolean(true);
+            loader.save(node);
+        } catch (ConfigurateException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index e079f4db4e4738f60a6fdbdbf5e4d1baf593a62f..7e0d2b3b93c4c0dd69ee9a3b29a280314e9e9349 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -644,7 +644,7 @@ public class ServerLevel extends Level implements WorldGenLevel, ca.spottedleaf.
 
         this.handlingTick = true;
         TickRateManager tickratemanager = this.tickRateManager();
-        boolean flag = tickratemanager.runsNormally();
+        boolean flag = !grinder.GrinderConfiguration.SERVER_SIDE_FROZEN && tickratemanager.runsNormally(); // Grinder
 
         if (flag) {
             gameprofilerfiller.push("world border");
@@ -759,7 +759,7 @@ public class ServerLevel extends Level implements WorldGenLevel, ca.spottedleaf.
             this.timings.entityTick.stopTiming(); // Spigot
             this.timings.tickEntities.stopTiming(); // Spigot
             gameprofilerfiller.pop();
-            this.tickBlockEntities();
+            if (grinder.GrinderConfiguration.BLOCK_ENTITY_TICKING) this.tickBlockEntities();
         }
 
         gameprofilerfiller.push("entityManagement");
