From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Sat, 22 Jun 2024 23:38:22 +0200
Subject: [PATCH] deactivate commands


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 65c0af9784a174b8432a61f8bf5a0a9adb69ff48..3aaca25772b894078194290d888b1e3dd5abd342 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -169,7 +169,7 @@ public class Commands {
         DefaultGameModeCommands.register(this.dispatcher);
         DifficultyCommand.register(this.dispatcher);
         EffectCommands.register(this.dispatcher, commandRegistryAccess);
-        EmoteCommands.register(this.dispatcher);
+        //EmoteCommands.register(this.dispatcher); // Grinder
         EnchantCommand.register(this.dispatcher, commandRegistryAccess);
         ExperienceCommand.register(this.dispatcher);
         FillCommand.register(this.dispatcher, commandRegistryAccess);
@@ -186,7 +186,7 @@ public class Commands {
         ListPlayersCommand.register(this.dispatcher);
         LocateCommand.register(this.dispatcher, commandRegistryAccess);
         LootCommand.register(this.dispatcher, commandRegistryAccess);
-        MsgCommand.register(this.dispatcher);
+        //MsgCommand.register(this.dispatcher); // Grinder
         ParticleCommand.register(this.dispatcher, commandRegistryAccess);
         PlaceCommand.register(this.dispatcher);
         PlaySoundCommand.register(this.dispatcher);
@@ -195,7 +195,7 @@ public class Commands {
         RecipeCommand.register(this.dispatcher);
         ReturnCommand.register(this.dispatcher);
         RideCommand.register(this.dispatcher);
-        SayCommand.register(this.dispatcher);
+        //SayCommand.register(this.dispatcher); // Grinder
         ScheduleCommand.register(this.dispatcher);
         ScoreboardCommand.register(this.dispatcher, commandRegistryAccess);
         SeedCommand.register(this.dispatcher, environment != Commands.CommandSelection.INTEGRATED);
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 663ae307bc84583ab271646e9b7fdb2eeeecc09c..79a1f71c125ad326d73f4bab71f82b91355bf12c 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -593,11 +593,11 @@ public final class CraftServer implements Server {
 
         if (type == PluginLoadOrder.POSTWORLD) {
             // Spigot start - Allow vanilla commands to be forced to be the main command
-            this.setVanillaCommands(true);
+            /*this.setVanillaCommands(true);
             this.commandMap.setFallbackCommands();
             this.setVanillaCommands(false);
             // Spigot end
-            this.commandMap.registerServerAliases();
+            this.commandMap.registerServerAliases();*/ // Grinder
             DefaultPermissions.registerCorePermissions();
             CraftDefaultPermissions.registerCorePermissions();
             if (!io.papermc.paper.configuration.GlobalConfiguration.get().misc.loadPermissionsYmlBeforePlugins) this.loadCustomPermissions(); // Paper
diff --git a/src/main/java/org/spigotmc/SpigotConfig.java b/src/main/java/org/spigotmc/SpigotConfig.java
index 38e8030d0d88def45e35de1df70da628d2bdaeaf..eabb67a08a520a5452f5c7fa7c6e8d7e044df84d 100644
--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -187,7 +187,7 @@ public class SpigotConfig
             }
         }
         SpigotConfig.tabComplete = SpigotConfig.getInt( "commands.tab-complete", 0 );
-        SpigotConfig.sendNamespaced = SpigotConfig.getBoolean( "commands.send-namespaced", true );
+        SpigotConfig.sendNamespaced = SpigotConfig.getBoolean( "commands.send-namespaced", false );
     }
 
     public static String whitelistMessage;
@@ -215,7 +215,7 @@ public class SpigotConfig
     }
 
     public static int timeoutTime = 60;
-    public static boolean restartOnCrash = true;
+    public static boolean restartOnCrash = false;
     public static String restartScript = "./start.sh";
     public static String restartMessage;
     private static void watchdog()
@@ -224,7 +224,7 @@ public class SpigotConfig
         SpigotConfig.restartOnCrash = SpigotConfig.getBoolean( "settings.restart-on-crash", SpigotConfig.restartOnCrash );
         SpigotConfig.restartScript = SpigotConfig.getString( "settings.restart-script", SpigotConfig.restartScript );
         SpigotConfig.restartMessage = SpigotConfig.transform( SpigotConfig.getString( "messages.restart", "Server is restarting" ) );
-        SpigotConfig.commands.put( "restart", new RestartCommand( "restart" ) );
+        //SpigotConfig.commands.put( "restart", new RestartCommand( "restart" ) ); // Grinder
         // WatchdogThread.doStart( SpigotConfig.timeoutTime, SpigotConfig.restartOnCrash ); // Paper - moved to after paper config initialization
     }
 
@@ -298,10 +298,10 @@ public class SpigotConfig
     public static List<String> spamExclusions;
     private static void spamExclusions()
     {
-        SpigotConfig.spamExclusions = SpigotConfig.getList( "commands.spam-exclusions", Arrays.asList( new String[]
+        SpigotConfig.spamExclusions = SpigotConfig.getList( "commands.spam-exclusions", /*Arrays.asList( new String[]
         {
                 "/skill"
-        } ) );
+        } )*/ List.of() ); // Grinder
     }
 
     public static boolean silentCommandBlocks;
@@ -310,7 +310,7 @@ public class SpigotConfig
         SpigotConfig.silentCommandBlocks = SpigotConfig.getBoolean( "commands.silent-commandblock-console", false );
     }
 
-    public static Set<String> replaceCommands;
+    /*public static Set<String> replaceCommands;
     private static void replaceCommands()
     {
         if ( SpigotConfig.config.contains( "replace-commands" ) )
@@ -320,7 +320,7 @@ public class SpigotConfig
         }
         SpigotConfig.replaceCommands = new HashSet<String>( (List<String>) SpigotConfig.getList( "commands.replace-commands",
                 Arrays.asList( "setblock", "summon", "testforblock", "tellraw" ) ) );
-    }
+    }*/ // Grinder
 
     public static int userCacheCap;
     private static void userCacheCap()
