From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Tue, 9 Jul 2024 00:59:33 +0200
Subject: [PATCH] better structure


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index dc03f0a871ff8a5e5219d8f5840cbcb12ffb6401..a98f45cccd47ea5288f1bb0868a8ae99d55a5038 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -124,7 +124,7 @@ public class Main {
                 this.acceptsAll(Main.asList("b", "bukkit-settings"), "File for bukkit settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("bukkit.yml"))
+                        .defaultsTo(new File(io.papermc.paper.configuration.PaperConfigurations.CONFIG_DIR, "bukkit.yml")) // Grinder - better structure
                         .describedAs("Yml file");
 
                 /*this.acceptsAll(Main.asList("C", "commands-settings"), "File for command settings")
@@ -153,7 +153,7 @@ public class Main {
                 this.acceptsAll(Main.asList("S", "spigot-settings"), "File for spigot settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("spigot.yml"))
+                        .defaultsTo(new File(io.papermc.paper.configuration.PaperConfigurations.CONFIG_DIR, "spigot.yml")) // Grinder - better structure
                         .describedAs("Yml file");
                 // Spigot End
 
