From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron <bauhd@gmx.net>
Date: Mon, 22 Jul 2024 23:27:24 +0200
Subject: [PATCH] remove update system


diff --git a/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java b/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java
index 70413fddd23ca1165cb5090cce4fddcb1bbca93f..c2427f995f2e0b87801a2c56e51fee5135a2a95e 100644
--- a/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java
+++ b/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java
@@ -43,11 +43,11 @@ public class PluginInitializerManager {
 
     private static PluginInitializerManager parse(@NotNull final OptionSet minecraftOptionSet) throws Exception {
         // We have to load the bukkit configuration inorder to get the update folder location.
-        final File configFileLocationBukkit = (File) minecraftOptionSet.valueOf("bukkit-settings");
+        //final File configFileLocationBukkit = (File) minecraftOptionSet.valueOf("bukkit-settings"); // Grinder - remove update system
 
         final Path pluginDirectory = ((File) minecraftOptionSet.valueOf("plugins")).toPath();
 
-        final YamlConfiguration configuration = PaperConfigurations.loadLegacyConfigFile(configFileLocationBukkit);
+        /*final YamlConfiguration configuration = PaperConfigurations.loadLegacyConfigFile(configFileLocationBukkit); // Grinder - remove update system
 
         final String updateDirectoryName = configuration.getString("settings.update-folder", "update");
         if (updateDirectoryName.isBlank()) {
@@ -81,7 +81,7 @@ public class PluginInitializerManager {
             return new PluginInitializerManager(pluginDirectory, null);
         }
 
-        return new PluginInitializerManager(pluginDirectory, resolvedUpdateDirectory);
+        return new PluginInitializerManager(pluginDirectory, resolvedUpdateDirectory);*/return new PluginInitializerManager(pluginDirectory, null); // Grinder - remove update system
     }
 
     public static PluginInitializerManager init(final OptionSet optionSet) throws Exception {
diff --git a/src/main/resources/configurations/bukkit.yml b/src/main/resources/configurations/bukkit.yml
index 27dc9b87e9ee86202cf16fee0937ebc19a0c3c0f..373f1ebdd2ba943e7a6bd2ef1c46b797b2451f5c 100644
--- a/src/main/resources/configurations/bukkit.yml
+++ b/src/main/resources/configurations/bukkit.yml
@@ -14,7 +14,6 @@
 settings:
     allow-end: false
     warn-on-overload: true
-    update-folder: update
     plugin-profiling: false
     connection-throttle: 4000
     query-plugins: true
